web_src/js/features/repo-issue.js

```diff
@@ -680,3 +688,50 @@ export function initIssueTemplateCommentEditors($commentForm) {
     initCombo($(el));
   }
 }
+
+function initRepoIssueStateButton() {
+  const $statusButton = $('#status-button');
+  if (!$statusButton.length) return;
+
+  const $statusDropdown = $('#status-dropdown');
+  const selectedValue = $statusDropdown.find('input[type=hidden]').val();
+
+  $statusButton.on('click', (e) => {
+    if ($statusDropdown.find('input[type=hidden]').val() !== '4') return;
+    // if click the button of "close as duplicate", show modal to let users select issue firstly.
+    e.preventDefault();
+    // save the original value of "type" and "state" input
+    const originalType = $('#type').val();
+    const originalSate = $('#state').val();
+    // temporarily reset the input value
+    $('#type').val('issue');
+    $('#state').val('all');
+    initIssueSearchDropdown('#duplicate-issues-list');
+    const $duplicateModal = $('#duplicate-issue-modal');
+    $duplicateModal.modal({
+      onHidden() { // close modal
+        // clear selected item in the dropdown
+        $('#duplicate-issues-list').dropdown('set exactly', []);
+        // restore the value of "type" and "state" input
+        $('#type').val(originalType);
+        $('#state').val(originalSate);
+      },
+      onApprove() {
+        $('#duplicate_issue_id').val($duplicateModal.find('input[type=hidden]').val());
+        const $form = $('#comment-form');
+        $form[0].dispatchEvent(new SubmitEvent('submit', {bubbles: true, cancelable: true, submitter: $statusButton[0]}));
```

Comment:
Use [form.submit](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/submit) if possible.

Reply:
I have tried to use `$form.trigger('sbumit')`, but it does not work well here.
Because this form has been added a class of `form-fetch-action`, if  just use `$form.trigger('sbumit')` to submit, we can't get the values of the `name` and `value` attributes which are bound to the button (see https://github.com/go-gitea/gitea/blob/d2c3a90ee24f6e9e3ffff164fa7e675a269e5add/web_src/js/features/common-global.js#L96).
So I think we should use `new SubmitEvent` to specify the `submitter` here.

Reply:
Hmm, did you try `$form[0].submit()` as well, or only the jQuery wrapper?